From 10838b66f4171b7a65a2130e4e72d200cfdff1fa Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Tue, 25 Nov 2025 16:56:01 +0100
Subject: [PATCH] usb: dwc2: hcd: re-enable interrupts upon suspend

Don't disable global interrupts during bus suspend. This can make
the controller not responsive with runtime PM bus suspend. The
original issue used to trigger irq loop, due to bad exit from
clock gating. It's now been fixed.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I742b4041c061a6ca378ef895ad63403deb7531cd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/512993
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/usb/dwc2/hcd.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/usb/dwc2/hcd.c b/drivers/usb/dwc2/hcd.c
index cbb5e7041b2c..2ea26e3c9eab 100644
--- a/drivers/usb/dwc2/hcd.c
+++ b/drivers/usb/dwc2/hcd.c
@@ -4392,7 +4392,6 @@ static int _dwc2_hcd_suspend(struct usb_hcd *hcd)
 		 * clock gating is used to save power.
 		 */
 		if (!hsotg->params.no_clock_gating) {
-			dwc2_disable_global_interrupts(hsotg);
 			dwc2_host_enter_clock_gating(hsotg);
 
 			/* After entering suspend, hardware is not accessible */
-- 
2.53.0

